tests/unit/test_utils.py

```diff
@@ -0,0 +1,85 @@
+"""Tests for the `utils` module."""
+
+from pathlib import Path
+from typing import Callable
+from unittest.mock import MagicMock, patch
+
+from autoimport.utils import get_pyproject_path, path_contains_pyproject
+
+
+class TestContainsPyproject:
+    """Tests for the `contains_pyproject` function"""
+
+    def test_pyproject_found(self, create_tmp_file: Callable) -> None:
+        """
+        Given: a path containing a `pyproject.toml` file,
+        When: the `contains_pyproject` function is invoked with the path,
+        Then: a `True` is returned
+        """
+        path = create_tmp_file(filename="pyproject.toml")
+
+        result = path_contains_pyproject(path.parent)
+
+        assert result is True
+
+    def test_pyproject_not_found(self, create_tmp_file: Callable) -> None:
+        """
+        Given: a path not containing a `pyproject.toml` file,
+        When: the `contains_pyproject` function is invoked with the path,
+        Then: a `False` is returned
+        """
+        path = create_tmp_file(filename="foo.toml")
+
+        result = path_contains_pyproject(path.parent)
+
+        assert result is False
+
+
+class TestGetPyprojectPath:
+    """Tests for the `get_pyproject_path`"""
+
+    @patch("autoimport.utils.Path")
+    def test_in_current_directory(
+        self, mock_path: MagicMock, create_tmp_file: Callable
+    ) -> None:
+        """
+        Given: a `pyproject.toml` file in the `cwd`,
+        When: the `get_pyproject_path` function is invoked,
+        Then: the path to the `pyproject.toml` is returned
+        """
+        path_to_pyproject = create_tmp_file(filename="pyproject.toml")
+        mock_path.cwd.return_value = path_to_pyproject.parent
+
+        result = get_pyproject_path()
+
+        assert result == path_to_pyproject
+
+    @patch("autoimport.utils.Path")
+    def test_in_parent_directory(
+        self, mock_path: MagicMock, create_tmp_file: Callable
+    ) -> None:
+        """
+        Given: a `pyproject.toml` file in the parent of `cwd`,
+        When: the `get_pyproject_path` function is invoked,
+        Then: the path to the `pyproject.toml` is returned
+        """
+        path_to_pyproject = create_tmp_file(filename="pyproject.toml")
+        sub_dir = path_to_pyproject / "sub"
+        mock_path.cwd.return_value = sub_dir
+
+        result = get_pyproject_path()
+
+        assert result == path_to_pyproject
+
+    @patch("autoimport.utils.Path")
+    def test_not_found(self, mock_path: MagicMock) -> None:
+        """
+        Given: no `pyproject.toml` in the `cwd` or parent dirs,
+        When: the `get_pyproject_path` function is invoked,
+        Then: `None` is returned
+        """
+        mock_path.cwd.return_value = Path("/nowhere")
+
+        result = get_pyproject_path()
+
+        assert result is None
```

Comment:
I'm missing the loading of the config from the [command line entrypoint](https://github.com/lyz-code/autoimport/blob/master/src/autoimport/entrypoints/cli.py#L18), I know we're not using it yet, and if you prefer to add that part of the code and their e2e tests on the next PR I'm ok with it

Reply:
yeah i thought that could come in the next pr, thought it would be useful to see how it's used in determining where it should be loaded

Reply:
Also I've found in the past that when I try to use it at entrypoint level I realize that the implementation I've done needs to be changed in some way that I have not foresaw. Your implementation looks flawless to me, but you know how development surprises you every time

Reply:
yeah sure 👍